File: javaDesde0/ejercicios1/ProyectoP1/TaxiCond.java
public class TaxiCond{

    private Taxi vehiculoTaxi;
    private Persona conductorTaxi;
    
    public TaxiCond(){
    
        vehiculoTaxi = new Taxi(); // creamos un objeto con el constructor general de Taxi
        conductorTaxi = new Persona(); // creamos un objeto Persona con el constructor general de Persona

    }
    
    public TaxiCond(Taxi objetoTaxi, Persona objetoPersona){
    
        vehiculoTaxi = new Taxi (objetoTaxi.getMatricula(),objetoTaxi.getDistrito(), objetoTaxi.getTipoMotor());
        conductorTaxi = new Persona(objetoPersona.getNombre() );
        
    }
    
    
    public void setMatricula(String valorMatricula){vehiculoTaxi.setMatricula(valorMatricula);}
    
    //Metodo que devuelve la informacion sobre el objeto TaxiCond
    
    public String getDatosTaxiCond(){
    
        String matricula = vehiculoTaxi.getMatricula();
        String distrito = vehiculoTaxi.getDistrito();
        int tipoMotor = vehiculoTaxi.getTipoMotor();
        String cadenaTipoMotor = "";
        
        if (tipoMotor == 0){cadenaTipoMotor = "desconocido";}
        else if (tipoMotor == 1){cadenaTipoMotor = "gasolina";}
        else if (tipoMotor == 2){cadenaTipoMotor = "disel";}
        
        String datosTaxiCond = "el objeto TaxiCond presenta estos datos. Matricula: " 
        + matricula + " Distrito "+distrito+" Tipo de motor: "+ cadenaTipoMotor;
        
        
        System.out.println(datosTaxiCond);
        return datosTaxiCond;
    }
   
}

File: ejercicios_Python_2/conversionPython/clases1/TaxiCond.py
from Persona import *
from Taxi import *

class TaxiCond:
    
    def __init__(self,taxi: Taxi, persona: Persona) :
        self.vehiculo_taxi = taxi   
        self.conductorTaxi = persona

    def getDatos(self):

        print("matricula con ", self.vehiculo_taxi.matricula)
        print("el distrito es ", self.vehiculo_taxi.distrito)
        print("con un tipo de motor", self.vehiculo_taxi.tipoMotor)      


taxicond = TaxiCond

taxi = Taxi(None,None,None,0)
taxi.Taxi(taxi.ciudad,taxi.matricula,taxi.distrito,taxi.tipoMotor)

persona = Persona(None,None,0,False,0)

persona.setNombre(persona.nombre)
taxi.setmatricula(taxi.matricula)
taxi.setDistrito(taxi.distrito)
taxi.setTipoMotor(taxi.tipoMotor)

print("El nombre del dueño del taxi es:",persona.nombre)
print("La matricula de el taxi es:",taxi.matricula)
print("El distrito donde se encuentra es:",taxi.distrito)
print("El tipo de motor que tienen es:",taxi.tipoMotor)


